tardis/visualization/tools/convergence_plot.py

```diff
@@ -0,0 +1,388 @@
+        self.value_data = defaultdict(list)
+        self.iterations = iterations
+        self.current_iteration = 1
+        self.luminosities = ["Emitted", "Absorbed", "Requested"]
+        self.plasma_plot = None
+        self.luminosity_plot = None
+
+        if "colorscale" in kwargs:
+            self.colorscale = transition_colors(
+                kwargs["colorscale"], iterations=self.iterations
+            )
+        else:
+            self.colorscale = transition_colors(iterations=self.iterations)
+
+        if "plasma_plot_config" in kwargs:
+            if not isinstance(kwargs["plasma_plot_config"], dict):
+                raise TypeError("Expected dict in plasma_plot_config argument")
+            self.plasma_plot_config = kwargs["plasma_plot_config"]
+
+        if "luminosity_plot_config" in kwargs:
+            if not isinstance(kwargs["luminosity_plot_config"], dict):
+                raise TypeError(
+                    "Expected dict in luminosity_plot_config argument"
+                )
+            self.luminosity_plot_config = kwargs["luminosity_plot_config"]
+
+    def fetch_data(self, name=None, value=None, item_type=None):
+        """
+        This allows user to fetch data from the Simulation class.
+        This data is stored and used when an iteration is completed.
+
+        Parameters
+        ----------
+        name : string
+            name of the data
+        value : string or array
+            string or array of quantities
+        item_type : string
+            either iterable or value
+
+        """
+        if item_type == "iterable":
+            self.iterable_data[name] = value
+        if item_type == "value":
+            self.value_data[name].append(value)
+
+    def create_plasma_plot(self):
+        """
+        Creates an empty plasma plot.
+        The default layout can be overridden by passing plasma_plot_config dictionary in the run_tardis function.
+        """
+        fig = go.FigureWidget().set_subplots(rows=1, cols=2, shared_xaxes=True)
+        fig.add_scatter(row=1, col=1)
+        fig.add_scatter(row=1, col=2)
+
+        fig = fig.update_layout(
+            xaxis={
+                "tickformat": "g",
+                "title": r"$\mbox{Velocity}~({km}~{s^{-1}})$",
+            },
+            xaxis2={
+                "tickformat": "g",
+                "title": r"$\mbox{Velocity}~({km}~{s^{-1}})$",
+                "matches": "x",
+            },
+            yaxis={
+                "tickformat": "g",
+                "title": r"$\mbox{T}_{rad}\ [K]$",
+                "tickmode": "auto",
+                "nticks": 15,
+            },
+            yaxis2={
+                "tickformat": "g",
+                "title": r"$\mbox{W}$",
+                "tickmode": "auto",
+                "nticks": 15,
+            },
+            height=450,
+            legend_title_text="Iterations",
+            margin=dict(l=10, r=135, b=25, t=25, pad=0),
+        )
+
```

Comment:
Label units should not be in italics. Also, the large capitals in T_rad and W should be italicised, but the subscripts should not be (same logic applies to all parameters, e.g. T_inner later).